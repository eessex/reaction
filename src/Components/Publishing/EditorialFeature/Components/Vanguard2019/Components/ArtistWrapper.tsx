import { Box, color, Flex, FlexProps, Sans, Serif } from "@artsy/palette"
import { Share, ShareContainer } from "Components/Publishing/Byline/Share"
import { getFullEditorialHref } from "Components/Publishing/Constants"
import { Emerging } from "Components/Publishing/EditorialFeature/Components/Vanguard2019/Blobs/Emerging"
import { GettingTheirDue } from "Components/Publishing/EditorialFeature/Components/Vanguard2019/Blobs/GettingTheirDue"
import { NewlyEstablished } from "Components/Publishing/EditorialFeature/Components/Vanguard2019/Blobs/NewlyEstablished"
import ArticleWithFullScreen from "Components/Publishing/Layouts/ArticleWithFullScreen"
import { StyledArtworkCaption } from "Components/Publishing/Sections/ArtworkCaption"
import { CaptionContainer } from "Components/Publishing/Sections/Caption"
import { FullScreenProvider } from "Components/Publishing/Sections/FullscreenViewer/FullScreenProvider"
import {
  Sections,
  StyledSections,
} from "Components/Publishing/Sections/Sections"
import { StyledText } from "Components/Publishing/Sections/StyledText"
import { ArticleData } from "Components/Publishing/Typings"
import { random } from "lodash"
import React from "react"
import styled from "styled-components"
import { slugify } from "underscore.string"

export interface SVGBackgroundProps extends FlexProps {
  url?: string
}
export class VanguardArtistWrapper extends React.Component<
  {
    article: ArticleData
    isExpanded?: boolean
    section?: string
  },
  {
    isExpanded: boolean
  }
> {
  state = {
    isExpanded: this.props.isExpanded || false,
  }

  componentDidMount() {
    const {
      article: { title },
    } = this.props
    const slug = slugify(title)

    if (window.location.pathname.includes(slug)) {
      this.onExpand()
    }
  }

  onExpand() {
    const { isExpanded } = this.state
    this.setState({ isExpanded: !isExpanded })
  }

  getRandomSVG(section) {
    /* Lodash random returns a whole integer between 0 and the max passed.
     * "Emerging" has 20 SVG possibilities while Newly Established
     * and "Getting Their Due" have 15, hence the default to 15
     * */
    switch (section) {
      case "emerging":
        return random(20)
      default:
        return random(15)
    }
  }

  getSVGBackground(index, section) {
    switch (section) {
      case "emerging":
        return Emerging(index)
      case "newly-established":
        return NewlyEstablished(index)
      case "getting-their-due":
        return GettingTheirDue(index)
    }
  }

  render() {
    const { article, section } = this.props
    const { hero_section, layout, slug, title } = article
    const { isExpanded } = this.state

    const background = this.getSVGBackground(
      this.getRandomSVG(section),
      section
    )
    const backgroundColor = isExpanded ? color("black100") : undefined

    return (
      <FullScreenProvider>
        <ArtistWrapper
          background={backgroundColor}
          pt={50}
          mb={isExpanded && 100}
          id={slugify(article.title)}
        >
          <BackgroundContainer>{background}</BackgroundContainer>
          <ArticleWithFullScreen article={article}>
            <ArtistContainer pb={4} maxWidth={1000} px={4} mx="auto">
              <Box textAlign="center" pb={30}>
                <ArtistTitle size="8">{title}</ArtistTitle>
                <Box position="absolute">
                  <Share
                    // TODO: We may need to use custom urls for in-page routing
                    url={getFullEditorialHref(layout, slug)}
                    title={title}
                    color={color("white100")}
                  />
                </Box>
                {hero_section && (
                  <InvertedSans size="4" weight="medium">
                    {hero_section.deck}
                  </InvertedSans>
                )}
              </Box>

              <Sections
                hideAds
                article={article}
                customWidth={920}
                isTruncatedAt={!isExpanded && 2}
              />

              <ReadMoreButton
                size="5"
                weight="medium"
                textAlign="center"
                onClick={this.onExpand.bind(this)}
              >
                {isExpanded ? "Close" : "Read More"}
                <Sans size="8">{isExpanded ? "\u2191" : "\u2193"}</Sans>
              </ReadMoreButton>
            </ArtistContainer>
          </ArticleWithFullScreen>
        </ArtistWrapper>
      </FullScreenProvider>
    )
  }
}

export const InvertedSerif = styled(Serif)`
  mix-blend-mode: difference;
  background: ${color("black100")};
  color: ${color("white100")};
`

export const InvertedSans = styled(Sans)`
  mix-blend-mode: difference;
  background: ${color("black100")};
  color: ${color("white100")};
`

const ArtistTitle = styled(InvertedSerif)`
  font-size: 95px;
  line-height: 1em;
`

const ReadMoreButton = styled(InvertedSans)<{ onClick: () => void }>`
  text-transform: uppercase;
  cursor: pointer;
`

const ArtistContainer = styled(Box)`
  position: relative;

  /* override feature text drop-caps */
  p:first-child::first-letter,
  .paragraph:first-child::first-letter {
    font-family: inherit;
    font-size: inherit;
    float: none;
    line-height: inherit;
    margin-right: 0;
    margin-top: 0;
    text-transform: none;
  }

  ${StyledSections} {
    margin-top: 0;
  }

  ${ShareContainer},
  ${StyledText} {
    mix-blend-mode: difference;
    background: ${color("black100")};
    color: ${color("white100")};
  }

  ${CaptionContainer} {
    p {
      mix-blend-mode: difference;
      background: ${color("black100")};
      color: ${color("white100")};
    }

    a {
      mix-blend-mode: difference;
      background: ${color("black100")};
      color: ${color("white100")};
      background-image: linear-gradient(
        to bottom,
        transparent 0,
        #fff 1px,
        transparent 0
      );
      background-size: 1.25px 4px;
      background-repeat: repeat-x;
      background-position: bottom;
    }
  }

  ${StyledArtworkCaption} {
    mix-blend-mode: difference;
    background: ${color("black100")};
    color: ${color("white100")};

    a {
      mix-blend-mode: difference;
      background: ${color("black100")};
      color: ${color("white100")};
    }
  }

  ${StyledText} {
    a {
      mix-blend-mode: difference;
      background: ${color("black100")};
      color: ${color("white100")};
      background-image: linear-gradient(
        to bottom,
        transparent 0,
        #fff 1px,
        transparent 0
      );
      background-size: 1.25px 4px;
      background-repeat: repeat-x;
      background-position: bottom;
    }
  }
`

const BackgroundContainer = styled(Box)`
  height: 100%;
  position: absolute;
  top: 0;
  left: 0;
  right: 0;
`

const ArtistWrapper = styled(Flex)`
  flex-direction: column;
  position: relative;
  padding-bottom: 100px;
  min-height: 100vh;
`
