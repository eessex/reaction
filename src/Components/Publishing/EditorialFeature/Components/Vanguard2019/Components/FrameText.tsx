import { Flex, Sans } from "@artsy/palette"
import React from "react"
import styled from "styled-components"
import { Media } from "Utils/Responsive"

export const VanguardFrameText: React.SFC = () => {
  return (
    <>
      <Media at="xs">
        <MobileFrame>
          <MobileFrameText size="8">The Artsy</MobileFrameText>
          <MobileFrameText size="8" isUppercase>
            Vanguard
          </MobileFrameText>
          <MobileFrameText size="8" isUppercase>
            2019
          </MobileFrameText>
        </MobileFrame>
      </Media>

      <Media greaterThan="xs">
        <FrameTextLeft size={["12", "12", "14", "16"]}>Vanguard</FrameTextLeft>
        <FrameTextRight size={["12", "12", "14", "16"]}>2019</FrameTextRight>
      </Media>
    </>
  )
}

const FrameText = styled(Sans)`
  position: fixed;
  top: 50%;
  text-transform: uppercase;
  z-index: 2;
`

const FrameTextLeft = styled(FrameText)`
  left: 0;
  transform: rotate(-90deg);
  transform-origin: 30% 136%;
`

const FrameTextRight = styled(FrameText)`
  right: 0;
  transform: rotate(90deg);
  transform-origin: 60% 60%;
`

const MobileFrame = styled(Flex)`
  position: fixed;
  flex-direction: column;
  top: 65px;
  left: 0;
  height: calc(100vh - 150px);
  width: 60px;
  justify-content: space-between;
`

const MobileFrameText = styled(Sans)<{ isUppercase?: boolean }>`
  transform: rotate(90deg);
  white-space: nowrap;
  padding: 0 10px;

  ${({ isUppercase }) =>
    isUppercase &&
    `
    text-transform: uppercase;
  `}
`
