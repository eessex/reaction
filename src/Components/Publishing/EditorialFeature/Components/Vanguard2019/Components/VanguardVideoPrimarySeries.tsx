import { Box, color } from "@artsy/palette"
import React from "react"
import styled from "styled-components"

interface Props {
  url: string
}

export const VanguardVideoPrimarySeries: React.SFC<Props> = props => {
  return (
    <PrimarySeriesWrapper>
      <svg
        viewBox="0 0 1600 900"
        xmlns="http://www.w3.org/2000/svg"
        fill={color("white100")}
        {...props}
      >
        <path d="M0,0V900H1600V0ZM1117.25,859.16c-321.9,34.84-538.72,61.19-699.67-12s-244.75-34.58-297.95-94.44S61.14,621.06,47.84,566.53-2.66,374.1,61.31,242.72c40.62-83.44,50.38-148.4,146.69-183,72.79-26.07,208.29,0,296.08,4,35.91,1.64,149-21.28,207.51-39.9s176.91-16,248.74,0,186.22-1.33,262,1.33,275.35,5.32,283.33,234.11-20,235.44,0,359.14S1439.15,824.32,1117.25,859.16Z" />
      </svg>
      <VanguardIntroVideoWrapper>
        <VanguardIntroVideo
          autoPlay
          loop
          muted
          playsInline
          controls={false}
          src={props.url}
        />
      </VanguardIntroVideoWrapper>
    </PrimarySeriesWrapper>
  )
}

const VanguardIntroVideoWrapper = styled(Box)`
  height: 100%;
  max-width: 100vw;
  position: absolute;
  top: 0;
  z-index: -1;
`

const VanguardIntroVideo = styled.video`
  max-height: 98%;
`

const PrimarySeriesWrapper = styled(Box)`
  position: absolute;
  top: 0;
  left: 0;
  right: 0;
  overflow: hidden;
`
