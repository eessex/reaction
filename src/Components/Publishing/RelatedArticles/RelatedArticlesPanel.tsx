import { Sans } from "@artsy/palette"
import { garamond } from "Assets/Fonts"
import { once } from "lodash"
import React from "react"
import Waypoint from "react-waypoint"
import styled from "styled-components"
import { crop } from "../../../Utils/resizer"
import { track } from "../../../Utils/track"
import { getArticleHref } from "../Constants"

interface RelatedArticlesPanelProps extends React.HTMLProps<HTMLDivElement> {
  label?: string
  tracking?: any
  articles: Array<{
    thumbnail_title: string
    thumbnail_image: string
    slug: string
    id: string
  }>
}

export class RelatedArticlesPanel extends React.Component<
  RelatedArticlesPanelProps
> {
  static defaultProps = {
    label: "Related Stories",
  }
  onClick = article => {
    const { tracking } = this.props
    const href = getArticleHref(article.slug)

    tracking.trackEvent({
      action: "Clicked article impression",
      destination_path: href,
      entity_id: article.id,
      entity_type: "article",
      flow: "article",
      impression_type: "Related article",
      type: "link",
    })
  }

  trackRelatedImpression = () => {
    const { tracking } = this.props

    tracking.trackEvent({
      action: "article_impression",
      impression_type: "Related articles",
    })
  }

  render() {
    const { articles, label } = this.props

    return (
      <RelatedArticlesContainer>
        <Label size="3t" weight="medium">
          {label}
        </Label>
        <Waypoint onEnter={once(this.trackRelatedImpression)} />
        <Collection>
          {articles.map((article, i) => {
            const href = getArticleHref(article.slug)
            const articleImageSrc = crop(article.thumbnail_image, {
              width: 160,
              height: 110,
            })

            return (
              <ArticleLink
                href={href}
                key={`relatedArticles-${i}`}
                onClick={() => this.onClick(article)}
              >
                <ArticleImage src={articleImageSrc} />
                <ArticleTitle>{article.thumbnail_title}</ArticleTitle>
              </ArticleLink>
            )
          })}
        </Collection>
      </RelatedArticlesContainer>
    )
  }
}

const RelatedArticlesContainer = styled.div`
  max-width: 360px;
`

const Collection = styled.div`
  display: flex;
  flex-direction: column;
`

const Label = Sans.extend`
  margin-bottom: 10px;
`

export const ArticleLink = styled.a`
  text-decoration: none;
  display: flex;
  justify-content: left;
  margin-bottom: 20px;

  &:last-child {
    margin-bottom: 0;
  }
`

const ArticleImage = styled.img`
  min-width: 80px;
  height: 55px;
  margin-right: 10px;
`

const ArticleTitle = styled.span`
  ${garamond("s17")};
  color: black;
`

export default track()(RelatedArticlesPanel)
