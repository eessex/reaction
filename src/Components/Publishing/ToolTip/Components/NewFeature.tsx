import styled from "styled-components"
import React from "react"
import { unica } from "Assets/Fonts"
import Colors from "Assets/Colors"

const FEEDBACK_EMAIL =
  "mailto:productfeedback@artsy.net?subject=Feedback on 'Article Tooltips'"

export const NewFeature: React.SFC = () => {
  return (
    <NewFeatureContainer>
      <span>This is a new feature. </span>
      <a href={FEEDBACK_EMAIL} target="_blank">
        Tell us what you think.
      </a>
    </NewFeatureContainer>
  )
}

export const NewFeatureContainer = styled.div`
  ${unica("s10")};
  color: ${Colors.grayMedium};
  a {
    color: ${Colors.grayMedium};
    border-bottom: 1px solid ${Colors.grayRegular};
    line-height: 1em;
  }
`
