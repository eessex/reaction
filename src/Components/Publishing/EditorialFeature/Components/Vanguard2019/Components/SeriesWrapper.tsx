import { Box, Flex, media, Sans, Serif } from "@artsy/palette"
import { Share } from "Components/Publishing/Byline/Share"
import { getFullEditorialHref } from "Components/Publishing/Constants"
import { VanguardVideoBackground } from "Components/Publishing/EditorialFeature/Components/Vanguard2019/Components/VanguardVideoBackground"
import { ArticleData } from "Components/Publishing/Typings"
import { times } from "lodash"
import React from "react"
import styled from "styled-components"
import { slugify } from "underscore.string"
import { VanguardArtistWrapper } from "./ArtistWrapper"

export const VanguardSeriesWrapper: React.SFC<{
  article: ArticleData
  index: number
}> = props => {
  const { article, index } = props
  const { relatedArticles, title, layout, series, slug } = article
  const slugifiedTitle = slugify(title)
  const { hero_section } = props.article
  const url = ((hero_section && hero_section.url) || "") as string
  const isVideo = url.includes("mp4")
  const getVideoID = section => {
    switch (section) {
      case "newly-established":
        return "#clip-svg-newly-established"
      case "emerging":
        return "#clip-svg-emerging"
      case "getting-their-due":
        return "#clip-svg-getting-their-due"
    }
  }

  return (
    <Box id={slugifiedTitle}>
      <Box
        pt={50}
        // prevents overlapping nav on jump-link
      />
      <Box height="95vh" mb={80}>
        {isVideo && (
          <VanguardVideoBackground id={getVideoID(slugifiedTitle)} url={url} />
        )}
        <Box mx="auto" maxWidth={["80vw", 1400]} px={4}>
          <Numeral size={["10", "10", "10", "12"]}>
            {times(index + 1, () => "I")}
          </Numeral>
          <Title
            size={["12", "12", "12", "16"]}
            textAlign="center"
            element="h2"
          >
            {title}
          </Title>
        </Box>
      </Box>
      <Box mx="auto" maxWidth="65%" px={4} pb={150}>
        <Flex flexDirection="column" alignItems="center">
          {series && (
            <SubTitle size="12" element="h3" pb={2}>
              {series.sub_title}
            </SubTitle>
          )}
          <Share
            // TODO: We may need to use custom urls here for in-page routing
            url={getFullEditorialHref(layout, slug)}
            title={title}
          />
        </Flex>
      </Box>
      {/** map sub-series artist articles */}
      {relatedArticles &&
        relatedArticles.map((artistArticle, i) => (
          <VanguardArtistWrapper
            key={i}
            article={artistArticle}
            section={slugifiedTitle}
          />
        ))}
    </Box>
  )
}

const Title = styled(Sans)`
  text-transform: uppercase;
  text-align: center;
`

const SubTitle = styled(Serif)`
  text-transform: uppercase;
  text-align: center;
`

const Numeral = styled(Serif)`
  position: absolute;
  line-height: 1em;

  ${media.xl`
    font-size: 100px;
  `}
`
