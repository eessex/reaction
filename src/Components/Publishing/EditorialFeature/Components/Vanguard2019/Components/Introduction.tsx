import { Box, Flex, Sans, Serif } from "@artsy/palette"
import { Byline, BylineContainer } from "Components/Publishing/Byline/Byline"
import { Text } from "Components/Publishing/Sections/Text"
import { ArticleData } from "Components/Publishing/Typings"
import React from "react"
import styled from "styled-components"
import { VanguardVideoPrimarySeries } from "./VanguardVideoPrimarySeries"

export const VanguardIntroduction: React.SFC<{
  article: ArticleData
}> = props => {
  const { description } = props.article.series
  const { hero_section } = props.article
  const url = ((hero_section && hero_section.url) || "") as string
  const isVideo = url.includes("mp4")

  return (
    <IntroContainer>
      <Box minHeight="calc(100vh - 50px)" mb={150} pt={50}>
        <VideoWrapper>
          {isVideo && <VanguardVideoPrimarySeries url={url} />}
        </VideoWrapper>
        <HeaderText pt={70} size="8" textAlign="center">
          The Artsy
        </HeaderText>
      </Box>
      <Box mx="auto" maxWidth={980} px={4}>
        <Flex flexDirection="column" alignItems="center" pb={50}>
          <Title size="12" element="h1" textAlign="center" pb={1}>
            The Artists To Know Right Now
          </Title>
          <Byline {...props} />

          <Box textAlign="center" pt={50}>
            <Sans size="3t" weight="medium">
              Video by Alex John Beck
            </Sans>
            <Sans size="3t" weight="medium">
              Video Editing by Nate DeYoung
            </Sans>
            <Sans size="3t" weight="medium">
              Interaction Design by Wax Studios
            </Sans>
          </Box>
        </Flex>

        <Box pb={12}>
          <Text layout="standard" html={description} width="800px" />
        </Box>
      </Box>
    </IntroContainer>
  )
}

const VideoWrapper = styled(Flex)`
  flex-direction: row;
  position: relative;
  padding-bottom: 100px;
  min-height: 100vh;
`

const HeaderText = styled(Sans)`
  font-size: 100px;
  position: absolute;
  top: 2vw;
  width: 100%;
`

const IntroContainer = styled(Box)`
  ${BylineContainer} {
    div::before {
      display: none;
    }
  }
`

const Title = styled(Serif)`
  text-transform: uppercase;
  font-size: 90px;
  line-height: 1em;
`
