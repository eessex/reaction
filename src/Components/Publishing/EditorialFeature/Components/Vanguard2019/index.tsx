import { Box } from "@artsy/palette"
import { EditorialFeaturesProps } from "Components/Publishing/EditorialFeature/EditorialFeature"
import { Nav, NavContainer } from "Components/Publishing/Nav/Nav"
import { last } from "lodash"
import React from "react"
import styled from "styled-components"
import { slugify } from "underscore.string"
import { VanguardFrameText } from "./Components/FrameText"
import { VanguardIntroduction } from "./Components/Introduction"
import { VanguardSeriesWrapper } from "./Components/SeriesWrapper"
import { VanguardTableOfContents } from "./Components/TableOfContents"

export class Vanguard2019 extends React.Component<EditorialFeaturesProps> {
  public validSlugs

  onChangeSection = slug => {
    const scrollTarget = document.getElementById(slug)
    scrollTarget &&
      scrollTarget.scrollIntoView({
        behavior: "smooth",
      })
  }

  componentDidMount() {
    this.getValidPaths()
    window.addEventListener("load", this.handleScrollOnLoad.bind(this))
  }

  componentWillUnmount() {
    window.removeEventListener("load", this.handleScrollOnLoad)
  }

  handleScrollOnLoad() {
    const scrollSlugs = window.location.pathname
      .split("/series/artsy-vanguard-2019/")
      .filter(({ length }) => length)
    const scrollToSlug = scrollSlugs.length && last(scrollSlugs)

    if (this.validSlugs.includes(scrollToSlug)) {
      this.onChangeSection(scrollToSlug)
    }
    // remove slug from pathname
    window.history.replaceState({}, "", "/series/artsy-vanguard-2019")
  }

  getValidPaths = () => {
    const validPaths = []

    this.props.article.relatedArticles.forEach(series => {
      // get subSeries slug
      validPaths.push(slugify(series.title))
      series.relatedArticles.map(artist => {
        // get artist slug
        validPaths.push(slugify(artist.title))
      })
    })
    this.validSlugs = validPaths
  }

  render() {
    const { article } = this.props
    const { relatedArticles } = article

    return (
      <VanguardWrapper>
        <Nav
          canFix
          color="black"
          backgroundColor="white"
          title={article.title}
        />
        {/** floating title text */}
        <VanguardFrameText />
        {/** header landing video & intro text */}
        <VanguardIntroduction article={article} />
        {/** table of contents */}
        <VanguardTableOfContents
          article={article}
          onChangeSection={this.onChangeSection}
        />

        {/** map 3 sub-series articles */}
        {relatedArticles &&
          relatedArticles.map((subSeries, i) => (
            <VanguardSeriesWrapper
              key={subSeries.id}
              article={subSeries}
              index={i}
            />
          ))}
      </VanguardWrapper>
    )
  }
}

const VanguardWrapper = styled(Box)`
  background: white;
  z-index: 0;
  position: relative;

  ${NavContainer} {
    position: fixed;
  }
`
