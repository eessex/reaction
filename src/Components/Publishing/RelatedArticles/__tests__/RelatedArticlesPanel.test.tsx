import { mount } from "enzyme"
import "jest-styled-components"
import React from "react"
import renderer from "react-test-renderer"
import Waypoint from "react-waypoint"
import { RelatedPanel } from "../../Fixtures/Components"
import { ArticleLink, RelatedArticlesPanel } from "../RelatedArticlesPanel"

describe("RelatedArticlesPanel", () => {
  const getWrapper = props => {
    return mount(<RelatedArticlesPanel {...props} />)
  }

  let props
  beforeEach(() => {
    props = {
      articles: RelatedPanel,
      tracking: {
        trackEvent: jest.fn(),
      },
    }
  })

  it("renders the related articles panel", () => {
    const related = renderer
      .create(<RelatedArticlesPanel articles={RelatedPanel} />)
      .toJSON()
    expect(related).toMatchSnapshot()
  })

  it("renders a default label", () => {
    const component = getWrapper(props)
    expect(component.text()).toMatch("Related Stories")
  })

  it("renders a provided label", () => {
    props.label = "The Best Stories"
    const component = getWrapper(props)
    expect(component.text()).toMatch("The Best Stories")
  })

  it("renders links", () => {
    const component = getWrapper(props)
    expect(component.find(ArticleLink)).toHaveLength(3)
  })

  it("tracks link clicks", () => {
    const component = getWrapper(props)
    component
      .find(ArticleLink)
      .at(0)
      .simulate("click")
    const trackCall = props.tracking.trackEvent.mock.calls[0][0]

    expect(trackCall.action).toBe("Clicked article impression")
    expect(trackCall.destination_path).toBe(
      "/article/artsy-editorial-15-top-art-schools-united-states"
    )
    expect(trackCall.entity_id).toBe("52d99185cd530e581300006c")
    expect(trackCall.entity_type).toBe("article")
    expect(trackCall.flow).toBe("article")
    expect(trackCall.impression_type).toBe("Related article")
    expect(trackCall.type).toBe("link")
  })

  it("Calls a tracking impression", () => {
    const component = getWrapper(props)
    component
      .find(Waypoint)
      .getElement()
      .props.onEnter()
    const trackCall = props.tracking.trackEvent.mock.calls[0][0]

    expect(trackCall.action).toBe("article_impression")
    expect(trackCall.impression_type).toBe("Related articles")
  })
})
