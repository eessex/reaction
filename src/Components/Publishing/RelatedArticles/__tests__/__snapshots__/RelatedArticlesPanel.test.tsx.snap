// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RelatedArticlesPanel renders the related articles panel 1`] = `
.c2 {
  font-family: Unica77LLWebMedium,'Helvetica Neue',Helvetica,Arial,sans-serif;
  font-weight: 500;
  font-size: 14px;
  line-height: 20px;
}

.c0 {
  max-width: 360px;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c1 {
  margin-bottom: 10px;
}

.c4 {
  -webkit-text-decoration: none;
  text-decoration: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: left;
  -webkit-justify-content: left;
  -ms-flex-pack: left;
  justify-content: left;
  margin-bottom: 20px;
}

.c4:last-child {
  margin-bottom: 0;
}

.c5 {
  min-width: 80px;
  height: 55px;
  margin-right: 10px;
}

.c6 {
  font-family: 'Adobe Garamond W08','adobe-garamond-pro','AGaramondPro-Regular','Times New Roman',Times,serif;
  font-size: 17px;
  line-height: 1.1em;
  -webkit-font-smoothing: antialiased;
  color: black;
}

<div
  className="c0"
>
  <div
    className="c1 c2"
    fontFamily={
      Object {
        "fontFamily": "Unica77LLWebMedium, 'Helvetica Neue', Helvetica, Arial, sans-serif",
        "fontWeight": 500,
      }
    }
    fontSize={14}
  >
    Related Stories
  </div>
  <span
    style={
      Object {
        "fontSize": 0,
      }
    }
  />
  <div
    className="c3"
  >
    <a
      className="c4"
      href="/article/artsy-editorial-15-top-art-schools-united-states"
      onClick={[Function]}
    >
      <img
        className="c5"
        src="https://d7hftxdivxxvm.cloudfront.net?resize_to=fill&src=https%3A%2F%2Fartsy-media-uploads.s3.amazonaws.com%2F4Tq-iYkN8dOpshFoKRXyYw%252Fcustom-Custom_Size___PoetterHall_Exterior%2Bcopy.jpg&width=160&height=110&quality=95"
      />
      <span
        className="c6"
      >
        The 15 Top Art Schools in the United States
      </span>
    </a>
    <a
      className="c4"
      href="/article/artsy-editorial-four-years-walter-de-marias-death-final-work-complete"
      onClick={[Function]}
    >
      <img
        className="c5"
        src="https://d7hftxdivxxvm.cloudfront.net?resize_to=fill&src=https%3A%2F%2Fartsy-media-uploads.s3.amazonaws.com%2F6IqxBTQCkExip2auQ7ZWCA%252FDEMAR-2011.0006-B.jpg&width=160&height=110&quality=95"
      />
      <span
        className="c6"
      >
        Four Years after Walter De Mariaâ€™s Death, His Final Work Is Complete
      </span>
    </a>
    <a
      className="c4"
      href="/article/artsy-editorial-french-art-history-in-a-nutshell"
      onClick={[Function]}
    >
      <img
        className="c5"
        src="https://d7hftxdivxxvm.cloudfront.net?resize_to=fill&src=https%3A%2F%2Fartsy-media-uploads.s3.amazonaws.com%2FlEcCm2XbfZ7bPAVgLlM21w%252Flarger-21.jpg&width=160&height=110&quality=95"
      />
      <span
        className="c6"
      >
        French Art History in a Nutshell
      </span>
    </a>
  </div>
</div>
`;
